contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,203 @@
+>
+>Miško described the idea behind the framework in an <a href="https://www.infoworld.com/article/2612801/javascript/whats-so-special-about-googles-angularjs.html">interview done in 2013</a>. 
+
+With Angular, designers can use HTML as the template language and it allows for the extension of HTML's syntax to convey the application's components effortlessly. Angular makes much of the code you would otherwise have to write, completely redundant.
+
+## What are the different versions of Angular?
+
+AngularJS or Angular1 is a JavaScript front-end framework developed by Google whereas Angular is a TypeScript-based open-source front-end web application platform led by the Angular Team at Google.
+There have been 5 versions of Angular so far, which are AngularJS, Angular2, Angular4, Angular5 and the latest one being Angular6.
+There are key differences between the various versions as Google tried to enhance the tool given the fierce competition from React and Vue. 
+More details about the differences can be found in this [Quora](https://www.quora.com/What-are-the-different-versions-of-AngularJS) post.
+
+
+
+## Why learn Angular?
+
+Angular is not the world’s simplest framework, and it takes time to understand some of the concepts that Angular has to offer.
+But once you do, there are some really powerful things you can do in a surprisingly small amount of code.
+
+1. **Angular supports Single Page Applications.** Single Page Applications are a type of web applications that load a single HTML page, and the page is updated dynamically according to the interaction of the user. SPAs can communicate with the back-end servers without refreshing the full web page, for loading data in the application. They provide better user experience and also eases the development process.
+                                                  The fact that Angular supports the development of SPAs, makes it worth learning!
+ 
+2. **Angular uses TypeScript.** Angular applications are built using TypeScript language, a superscript for JavaScript, which ensures higher security as it supports types (primitives, interfaces, etc.). It helps catch and eliminate errors early when writing the code or performing maintenance tasks. This language ensures improved navigation, refactoring, and autocompletion services thereby easing the development process for engineers. You can even opt out of its inbuilt features when needed.
+TypeScript was created and is maintained by Microsoft, and has been <a href="https://developer.telerik.com/featured/the-rise-of-typescript/">growing in popularity</a> for the last few years; therefore it’s not much of a risk to incorporate TypeScript in to your application development. It’s not going anywhere.
```

Comment:
extra space to separate the text? If not it is continuous in one chunk when I check it out in `markdown`